CONTRIBUTING_GUIDELINES.md

```diff
@@ -0,0 +1,157 @@
+  * Use the body to explain _what_ and _why_ instead of _how_
+* Has been signed by the author ([see below](#certificate-of-origin---sign-your-work)).
+
+## License
+
+By contributing your code, you agree to license your contribution under the
+terms of the [Apache License](./LICENSE).
+
+If you are adding a new file it should have a header like below. In some
+languages, e.g. Python, you may need to change the comments to start with `#`.
+The easiest way is to copy the header from one of the existing source files and
+make sure the year is current and the copyright says "The Jaeger Authors".
+
+```
+// Copyright (c) 2018 The Jaeger Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+```
+
+## Certificate of Origin - Sign your work
+
+By contributing to this project you agree to the
+[Developer Certificate of Origin](https://developercertificate.org/) (or simply
+[DCO](./DCO)). This document was created by the Linux Kernel community and is a
+simple statement that you, as a contributor, have the legal right to make the
+contribution.
+
+The sign-off is a simple line at the end of the explanation for the patch, which
+certifies that you wrote it or otherwise have the right to pass it on as an
+open-source patch. The rules are pretty simple: if you can certify the
+conditions in the [DCO](./DCO), then just add a line to every git commit
+message:
+
+    Signed-off-by: Bender Bending Rodriguez <bender.is.great@gmail.com>
+
+using your real name (sorry, no pseudonyms or anonymous contributions.) You can
+add the sign off when creating the git commit via `git commit -s`.
+
+If you want signing to be automatic you can set up some aliases:
+
+### Missing sign-offs
+
+Note that **every commit in the pull request must be signed**. Jaeger
+repositories are configured with a [DCO-bot][dco-bot] that will check sign-offs
+on every commit and block the PR from being merged if some commits are missing
+sign-offs. If you only have one commit or the latest commit in the PR is missing
+a sign-off, the simplest way to fix this is to run:
+
+```
+git commit --amend -s
+```
+
+which will prompt you to edit the commit message while adding a signature.
+Simply accept the text as is, and push the branch:
+
+```
+git push --force
+```
+
+If some commit in the middle of your commit history is missing the sign-off, the
+simplest solution is to squash the commits into one and sign it. For example,
+suppose that your branch history looks like this:
+
+```
```

Comment:
This file is currently a straight copy of the jaeger repo version, which makes updating eaiser - but means the list of commits here given as example are related to that repo. Not sure it matters as just used for illustration.

Reply:
I don't think it matters.